woocommerce/woocommerce-ios
# [Mobile Payments] Show "Both WCPay and Stripe Extension are active" screen when both plugins installed

Implements a first draft of #5528 

This PR unblocks #5538 and ##5540 and would start allowing us to close the loop and test the whole integration.

There is no finalized design for the new screen, so I have put something together so that it provides enough UI to unblock progress building the rest of the integration.



### Description

* Add a new view to inform merchants that both WCPay and the Stripe extension are installed, and allow merchants to deactivate either of them
* Add logic to CardPresentPaymentsOnboardingUseCase so that we only check for the existence of both plugins once in the onboarding flow. 
* Add logic to save and load a boolean flag indicating whether the merchant has selected the Stripe Extension to user defaults.

Notes:
I asked for clarification on the following use case (see pdfdoF-fW#comment-837-p2):
Let’s assume a merchant has never setup In-Person Payments, and has both WCPay and the Stripe Extension installed and activated.

* The first time the merchant goes into In-Person Payments > Manage Card Reader, we should offer the merchant the screen where they can deactivate one of the plugins.
* The onboarding flow continues, the merchant completes the flow and is able to capture In-Person Payments.
* What happens if the merchant navigates to In-Person Payments > Manage Card Reader again? Are we displaying the same screen again, where they can deactivate one of the plugins?

In this PR I am submitting now, the merchant would be prompted, again, to deactivate one of the extensions. We should need to clarify whether that's the expected behaviour, but it does match the flowchart in pdfdoF-fW#comment-779-p2

We can refine this behaviour in a follow-up PR.

### Testing instructions

On a store with both WCPay and the Stripe Extension installed, navigate to Settings > In-Person Payments.
* If both are installed, you should be prompted to deactivate one of them
* It is possible to tap Refresh without having deactivated a plugin. We are not sure how this view is going to look yet, so I don't think it makes sense to add logic that we might need to remove next week. 
* After tapping "Refresh", the onboarding flow should continue, and should not prompt again to deactivate a plugin. 
* A breakpoint in AppSettingsStore.setStripeExtensionAvailability can help validate that the value saved to user defaults matched what was selected in the UI
* 
<img width="350" alt="Screen Shot 2022-01-17 at 3 14 30 PM" src="https://user-images.githubusercontent.com/2722505/149832167-13ebda64-24f7-421f-9b16-63deb34e9773.png">


### Screenshots



https://user-images.githubusercontent.com/2722505/149828425-bfd13657-7e40-452a-9be0-f1057d645bcf.mov


---
